app/controllers/admin/borrow_requests_controller.rb

```diff
@@ -2,7 +2,26 @@ class Admin::BorrowRequestsController < Admin::AdminController
   before_action :check_admin_logged_in, only: %i(index show edit)
 
   def index
-    @pagy, @borrow_reqs = pagy BorrowRequest.ordered_by_status
+    keyword, sort_kw = params.values_at :keyword, :sort_by
+
+    # Check if sort keyword doesn't exists then use default instead
+    sort_kw = "created_at" if sort_kw.blank?
+    @borrow_reqs = BorrowRequest.ordered_by sort_kw
+
+    # Collect "status"="1" into array of statuses
+    # params = {"status_pending": "0", "status_ready" : "1", ..}
+    # => statuses = ["ready", ..]
+    statuses = []
+    %w(pending ready borrowed returned rejected cancelled).each do |stt|
+      status_name = "status_#{stt}"
+      statuses << stt if params[status_name] == "1"
+    end
+    @borrow_reqs = @borrow_reqs.filter_by_status statuses if statuses.present?
+
+    # Seach for book title and author name
+    @borrow_reqs = @borrow_reqs.search_by keyword if keyword.present?
```

Comment:
```suggestion
    @borrow_reqs = @borrow_reqs.filter_by_status(statuses).search_by keyword
```
Xử lý parram có present hay ko thì đưa vào scope nhé

Reply:
Dạ chị